hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/s3guard/S3Guard.java

```diff
@@ -534,12 +530,18 @@ public TtlTimeProvider(long authoritativeDirTtl) {
       this.authoritativeDirTtl = authoritativeDirTtl;
     }
 
+    public TtlTimeProvider(Configuration conf) {
+      this.authoritativeDirTtl =
```

Comment:
Switch to Configuration.getTimeDuration(). Also, I think we should make the default time seconds, not millis. Just because our internal clocks are millis doesn't mean it makes a good configuration parameter for something which should always be measured in seconds. And if they want to switch to millis, that call makes it easy.